scenes/world_map/teleporter.gd

```diff
@@ -1,17 +1,72 @@
+@tool
 class_name Teleporter
 extends Area2D
 
-@export_file("*.tscn") var scene_to_go_to: String
+const SPAWN_POINT_GROUP_NAME: String = "spawn_point"
+
+@export_file("*.tscn") var scene_to_go_to: String:
+	set(new_value):
+		scene_to_go_to = new_value
+		_update_available_spawn_points()
+		notify_property_list_changed()
+
+var spawn_point_path: NodePath:
+	set(new_val):
+		if new_val == ^"NONE":
+			spawn_point_path = ^""
+		else:
+			spawn_point_path = new_val
+
+var _available_spawn_points: Array[NodePath] = []
 
 
 func _ready() -> void:
-	self.body_entered.connect(_on_body_entered)
 	collision_layer = 0
 	collision_mask = 0
 	set_collision_layer_value(3, true)
 	set_collision_mask_value(1, true)
 
+	if Engine.is_editor_hint():
+		return
+
+	# This is to stop a possible infinite loop if the player spawns on a
+	# teleporter that returns them to the teleporter that sent them
+	await get_tree().physics_frame
```

Comment:
Oh sorry, this comment was supposed to be part of my review!:

I'm not sure what case is being handled by this, but if teleporters and spawn points overlap, we can get a infinite portal loop (!)

https://github.com/user-attachments/assets/9fe40023-5c26-4b54-ab9d-9fc960f8eef9

Reply:
Heh. Honestly, I didn't test this. I assumed that waiting for a physics frame before connecting the signal should prevent the loops. I'll see what I can do

Reply:
It's a bit more complicated than it should. Created #56 to follow up on this